src/files.cpp

```diff
@@ -48,45 +52,131 @@ void glLoadTexture(SDL_Surface* image, int texnum)
 }
 
 
-bool completePath(char *dest, const char * const filename) {
-	if (!(filename && filename[0])) {
+static bool completeDataPath(char *dest, const char * const filename)
+{
+	if (!(filename && filename[0]))
+	{
 		return false;
 	}
 
 	// Already absolute
-	if (filename[0] == '/') {
+	if (filename[0] == '/')
+	{
 		strncpy(dest, filename, 1024);
 		return true;
 	}
-
 	snprintf(dest, 1024, "%s/%s", datadir, filename);
 	return true;
 }
 
-FILE* openDataFile(const char * const filename, const char * const mode) {
+int makeDirsRecursive(const char * path)
+{
+	const char * copying = path;
+	char soFar[1024] = {0};
+	while (*copying)
+	{
+		char cur = *copying;
+		soFar[copying - path] = *copying;
+		if (cur == '/' && mkdir(soFar, 0700) != 0 && errno != EEXIST && errno != EISDIR)
```

Comment:
Windows port note: This needs to be modified for Windows. Windows's `mkdir()` function is included in `direct.h` and doesn't take the permissions parameter.